internal/replay_delete.go

```diff
@@ -0,0 +1,114 @@
+package internal
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/mitchellh/colorstring"
+	"github.com/spf13/cobra"
+)
+
+// AddDeleteCommand returns cobra command delete, allows to delete a replay from a queue.
+func (r *ReplayCmd) AddDeleteCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete <slo-name>",
+		Short: "Delete a replay from a queue",
+		Long:  "Delete a replay from a queue.",
+		Args:  r.deleteArguments,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if r.project != "" {
+				r.client.Config.Project = r.project
+			}
+			if r.all {
+				return r.deleteAllReplays(cmd)
+			} else {
+				return r.deleteReplaysForSLO(cmd, r.sloName)
+			}
+		},
+	}
+
+	cmd.Flags().StringVarP(&r.project, "project", "p", "",
+		`Specifies the Project of the SLO you want to remove Replays from queue for.`)
+	cmd.Flags().BoolVar(&r.all, "all", false, "Delete ALL replays in queue.")
+
+	return cmd
+}
+
+func (r *ReplayCmd) deleteArguments(cmd *cobra.Command, args []string) error {
+	if !r.all && len(args) == 0 {
+		_ = cmd.Usage()
+		return errReplayDeleteInvalidOptions
+	}
+	if len(args) > 1 {
+		return errReplayTooManyArgs
+	}
+	if len(args) == 1 {
+		r.sloName = args[0]
+	}
+	return nil
+}
+
+type deleteReplayRequest struct {
+	Project string `json:"project,omitempty"`
+	Slo     string `json:"slo,omitempty"`
+	All     bool   `json:"all,omitempty"`
+}
+
+func (r *ReplayCmd) deleteAllReplays(cmd *cobra.Command) error {
+	cmd.Println(colorstring.Color("[yellow]Deleting all replays from queue[reset]"))
```

Comment:
I'd stick to capital letter as this is a proper name. This applies to other messages too :)

```suggestion
	cmd.Println(colorstring.Color("[yellow]Deleting all Replays from the queue[reset]"))
```